.github/workflows/pr.yml

```diff
@@ -71,26 +64,102 @@ jobs:
 
       - name: Build - IL2CPU
         run: |
-          dotnet publish IL2CPU\source\IL2CPU -r win7-x86 -o "$($env:USER_KIT_PATH)Build\IL2CPU\"
+          dotnet publish IL2CPU\source\IL2CPU -o "$($env:USER_KIT_PATH)Build\IL2CPU\"
 
       - name: Copy Output - Cosmos
         run: |
           xcopy /Y "Cosmos\source\Cosmos.Build.Tasks\tools\grub2\boot\grub\i386-pc\*.*" "$($env:USER_KIT_PATH)Build\ISO\boot\grub\i386-pc\"
           xcopy /Y "Cosmos\Build\grub2\boot\grub\grub.cfg" "$($env:USER_KIT_PATH)Build\ISO\boot\grub"
           xcopy /Y /S "Cosmos\Build\VMware\*" "$($env:USER_KIT_PATH)Build\VMware\"
           xcopy /Y /S "Cosmos\Build\Tools" "$($env:USER_KIT_PATH)Build\Tools\"
-          xcopy /Y /S "Cosmos\Resources\Bochs\*.*" "%ProgramFiles(x86)%\Bochs-2.6.8\"
 
       - name: Copy Output - IL2CPU
         run: |
           xcopy /Y "IL2CPU\source\Cosmos.Core.DebugStub\*.xs" "$($env:USER_KIT_PATH)XSharp\DebugStub\"
-          
-      - name: Test - Cosmos
+
+      # - name: Send test kernels to GITHUB_OUTPUT
+      #   id: testkernels_output
+      #   run: |
+      #     $testList = dotnet test Cosmos\Tests\Cosmos.TestRunner.UnitTest\Cosmos.TestRunner.UnitTest.csproj --list-tests
+      #     Write-Host $testList
+      #     $testList >> tests.txt
+      #     $match = Select-String -Pattern "^.*TestKernel\((.*)\)$" -Path tests.txt
+      #     Write-Host $match
+      #     $testKernels = $match.matches.groups | where { $_.index % 2 -ne 0 } | Select-Object value | foreach { $_.value }
+      #     Write-Host $testKernels
+      #     $testKernelsEnv = ""
+      #     $testKernels | foreach { $testKernelsEnv += "$($_)," }
+      #     $testKernelsEnv = $testKernelsEnv.Trim(",")
+      #     Write-Host $testKernelsEnv
+      #     "testkernels=[$($testKernelsEnv)]" >> $env:GITHUB_OUTPUT
+
+  test:
+    if: ${{ success() }}
+    needs: [build]
+    runs-on: self-hosted
+    
+    strategy:
+      matrix:
+        kernel: [
```

Comment:
we should add a note somewhere that when you make new tests you need to add them here